atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2378,6 +2364,37 @@ private boolean rollbackOtherTransaction(long startTs, TransactionService transa
         }
     }
 
+    ///////////////////////////////////////////////////////////////////////////
+    /// Post-commit actions
+    ///////////////////////////////////////////////////////////////////////////
+
+    private void runPostCommitActions() {
+        // Run close() to release locks before running success callbacks, since success callbacks might
+        // start a new transaction and attempt to grab the same locks as the current transaction.
+        close();
```

Comment:
So this is a slight behavior change: previously we'd run the callbacks, before calling close.

Reply:
Yes, it was a change in this instance, but this is actually what we want, see prior `AbstractTransactionManager L83-85` + https://github.com/palantir/atlasdb/pull/6191 - we want to run success callbacks before close in general.
```
            callback.run();
            txn.reportExpectationsCollectedData();
            txn.runSuccessCallbacksIfDefinitivelyCommitted();
```

Reply:
I agree, was calling out that this is a change.